projects/lib/src/board/board.h

```diff
@@ -601,7 +605,17 @@ inline const Move& Board::lastMove() const
 	return m_moveHistory.last().move;
 }
 
-inline bool Board::pieceHasMovement(int pieceType, unsigned movement) const
+inline bool Board::pieceHasCaptureMovement(Piece piece, int square, unsigned movement) const
+{
+    return Board::pieceHasMovement(piece.type(), square, movement);
+}
+
+inline bool Board::pieceHasMovement(Piece piece, int square, unsigned movement) const
+{
+    return Board::pieceTypeHasMovement(piece.type(), movement);
```

Comment:
Maybe add `Q_UNUSED(square)` here.

Reply:
There are more classes with unused arguments and the new `ShogiBoard`(s).. You might leave out the `square` identifier and only pass the formal argument `int` instead of `int square`.